src/sent.rs

```diff
@@ -153,10 +169,49 @@ impl SentPackets {
         self.congestion_ctrl.on_transmit(seq_num, transmit).unwrap();
     }
 
+    /// Handle an ACK for a packet with sequence number `ack_num`, and an optional selective_ack.
+    ///
+    /// Returns Error, if `ack_num` is not within the sequence number range of the sent packets.
+    pub fn on_ack(
+        &mut self,
+        ack_num: u16,
+        selective_ack: Option<&SelectiveAck>,
+        delay: Duration,
+        now: Instant,
+    ) -> Result<(CircularRangeInclusive, Vec<u16>), SentPacketsError> {
```

Comment:
is there a reason to separate the packets acknowledged by the `ack_num` and those acknowledged by the selective ACK?

Reply:
Mainly that the range is an efficient way to refer to many acknowledged sequence numbers, but the selective acks can't generally be encoded that way.

Maybe we could chain together an efficient range iterator, with an arbitrary vector of u16's, and return the iterator. Though I'm not really sure how to implement that in Rust, yet.